app/routes/_index.tsx

```diff
@@ -1,16 +1,102 @@
 import type { V2_MetaFunction } from '@remix-run/react';
 import HeroSection from '~/components/hero-section';
-import EventsSection from '../components/events-section';
+import { EventsSection } from '../components/events-section';
 
 export const meta: V2_MetaFunction = () => {
   return [{ title: 'Plan It Social' }];
 };
 
+type Event = {
+  id: number;
+  imgUrl: string;
+  imgAlt: string;
+  groupName: string;
+  eventTitle: string;
+  eventDescription: string;
+  date: string;
+  time: string;
+  location: string;
+  category: string;
+};
+
+export type EventsSectionProps = {
```

Comment:
Curious, why did you decide to declare the component prop types here? I haven't seen this pattern before. I would just keep them in the same file as the component. This keeps the types nicely co-located with its usage.

Reply:
I thought we will use these types in other routes like search.tsx. That's why kept all types in one place. This pattern was more like an experiment and it worked. Moving types to the components sound reasonable. Thanks. I'll do it!